src/test/java/nextstep/courses/domain/session/SessionTest.java

```diff
@@ -0,0 +1,149 @@
+        session.enroll(NsUserTest.JAVAJIGI,
+            new Payment(UUID.randomUUID().toString(), session.getSessionId(),
+                NsUserTest.JAVAJIGI.getId(), 800_000L));
+        session.enroll(NsUserTest.SANJIGI,
+            new Payment(UUID.randomUUID().toString(), session.getSessionId(),
+                NsUserTest.SANJIGI.getId(), 800_000L));
+
+        assertThatThrownBy(() -> {
+            session.enroll(NsUserTest.PARK,
+                new Payment(UUID.randomUUID().toString(), session.getSessionId(),
+                    NsUserTest.PARK.getId(), 800_000L));
+        }).isInstanceOf(CannotEnrollException.class);
+    }
+
+    @Test
+    @DisplayName("강의는 중복 수강신청할 수 없다")
+    void session_duplicate_student() throws CannotEnrollException, InvalidSessionDateException {
+        Session session = new Session(1L, SessionType.PAID, Status.OPEN, 800_000L, image, 10, date);
+        session.enroll(NsUserTest.JAVAJIGI,
+            new Payment(UUID.randomUUID().toString(), session.getSessionId(),
+                NsUserTest.JAVAJIGI.getId(), 800_000L));
+        session.enroll(NsUserTest.SANJIGI,
+            new Payment(UUID.randomUUID().toString(), session.getSessionId(),
+                NsUserTest.SANJIGI.getId(), 800_000L));
+        session.enroll(NsUserTest.PARK,
+            new Payment(UUID.randomUUID().toString(), session.getSessionId(),
+                NsUserTest.PARK.getId(), 800_000L));
+
+        assertThatThrownBy(() -> {
+            session.enroll(NsUserTest.SANJIGI,
+                new Payment(UUID.randomUUID().toString(), session.getSessionId(),
+                    NsUserTest.SANJIGI.getId(), 800_000L));
+        }).isInstanceOf(CannotEnrollException.class);
+    }
+
+    @Test
+    @DisplayName("강의 수강신청은 강의 상태가 모집중일 때만 가능하다")
+    void session_open_status() throws InvalidSessionDateException {
+        Session sessionPreparing = new Session(1L, SessionType.PAID, Status.PREPARING, 800_000L,
+            image, 10,
+            date);
+        assertThatThrownBy(() -> {
+            sessionPreparing.enroll(NsUserTest.JAVAJIGI,
+                new Payment(UUID.randomUUID().toString(), sessionPreparing.getSessionId(),
+                    NsUserTest.JAVAJIGI.getId(), 800_000L));
+        }).isInstanceOf(CannotEnrollException.class);
+
+        Session sessionClose = new Session(2L, SessionType.PAID, Status.CLOSE, 800_000L, image, 10,
+            date);
+        assertThatThrownBy(() -> {
+            sessionClose.enroll(NsUserTest.JAVAJIGI,
+                new Payment(UUID.randomUUID().toString(), sessionClose.getSessionId(),
+                    NsUserTest.JAVAJIGI.getId(), 800_000L));
+        }).isInstanceOf(CannotEnrollException.class);
+    }
+
+    @Test
+    @DisplayName("유료 강의는 수강생이 결제한 금액과 수강료가 일치할 때 수강 신청이 가능하다")
+    void session_paid_session() throws InvalidSessionDateException {
+        Session session = new Session(1L, SessionType.PAID, Status.OPEN, 800_000L, image, 2, date);
```

Comment:
```suggestion
        Session session = 유료_Session_생성됨(1L,800_000L,image,2,date);
```

반복되는 생성코드를 위해 별도 메소드로 분리시켜 보는것도 방법이겠네요~